static/js/publisher/release/actions/releases.js

```diff
@@ -1,5 +1,230 @@
+import "whatwg-fetch";
+
 export const UPDATE_RELEASES = "UPDATE_RELEASES";
 
+import { RISKS_WITH_AVAILABLE as RISKS } from "../constants";
+
+import { hideNotification, showNotification } from "./globalNotification";
+import { cancelPendingReleases } from "./pendingReleases";
+import { releaseRevisionSuccess, closeChannelSuccess } from "./channelMap";
+import { updateRevisions } from "./revisions";
+
+import { getRevisionsMap, initReleasesData } from "../releasesState";
+
+const ERROR_MESSAGE =
+  "There was an error while processing your request, please try again later.";
+
+function updateReleasesData(dispatch, releasesData) {
+  // init channel data in revisions list
+  const revisionsMap = getRevisionsMap(releasesData.revisions);
+  initReleasesData(revisionsMap, releasesData.releases);
+
+  dispatch(updateRevisions(revisionsMap));
+  dispatch(updateReleases(releasesData.releases));
+}
+
+function fetchReleasesHistory(csrfToken, snapName) {
+  return fetch(`/${snapName}/releases/json`, {
+    method: "GET",
+    mode: "cors",
+    cache: "no-cache",
+    credentials: "same-origin",
+    headers: {
+      "Content-Type": "application/json; charset=utf-8",
+      "X-CSRFToken": csrfToken
+    },
+    redirect: "follow",
+    referrer: "no-referrer"
+  })
+    .then(response => response.json())
+    .catch(() => {
+      throw new Error(ERROR_MESSAGE);
+    });
+}
+
+function fetchUpdatedReleasesHistory(dispatch, csrfToken, snapName) {
```

Comment:
I think this one is not needed as separate function, especially now when `dispatch` needs to be passed in.
All it does is just single `then`. We can do it inline where it's called. Especially that it seems we do it in one place anyway.